### PR TITLE
SDEMO-122: Startup blocks page (in progress)

### PR DESCRIPTION
**Pull request by thisisannie**
## [SDEMO-122](https://silverstripe.atlassian.net/browse/SDEMO-122) Startup Blocks page (with optional intro area)

Apps
Description

Template with elemental area and optional hero area that displays page title and breadcrumbs

AC’s

Checkbox to display the hero, will default to yes.

Hero area to display page title and breadcrumbs

Add elemental area

### CODE REVIEW

File: src/PageTypes/BlocksPage.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        'ShowHero' => true,
+    ];
+
+    public function getCMSFields()
+    {
+        $fields = parent::getCMSFields();
+
+        $fields->insertAfter(
+            'MenuTitle',
+            $metaTitle = CheckboxField::create(
```
Comment (TheSceneman) [inline src/PageTypes/BlocksPage.php:26]: This shouldnt be a named variable becuase we aren't using it anywhere else
```suggestion
            CheckboxField::create(
```

File: src/PageTypes/BlocksPage.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace SilverStripe\StartupThemeComponents\PageTypes;
+
+use Page;
+use SilverStripe\Forms\CheckboxField;
+
+class BlocksPage extends Page
+{
```
Comment (TheSceneman) [inline src/PageTypes/BlocksPage.php:9]: We should have a line after the opening brace and before the closing brace

File: src/PageTypes/BlocksPage.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    public function getCMSFields()
+    {
+        $fields = parent::getCMSFields();
+
+        $fields->insertAfter(
+            'MenuTitle',
+            $metaTitle = CheckboxField::create(
+                'ShowHero', 
+                'Show Hero',
+            )->setDescription('Show hero area containing breadcrumb and page name.'),
```
Comment (TheSceneman) [inline src/PageTypes/BlocksPage.php:29]:
```suggestion
            )->setDescription('Show hero area containing breadcrumbs and page name.'),
```

**Comment by thisisannie** [comment]
Related PR for styles: https://github.com/silverstripeltd/startup-theme/pull/12

**Review by TheSceneman — CHANGES_REQUESTED** [review]
Well done, a couple very minor tidy ups

File: src/PageTypes/BlocksPage.php (hunk 2 above)
Comment (thisisannie) [reply src/PageTypes/BlocksPage.php:9]: Will do!

**Review by TheSceneman — APPROVED** [review]
LGTM

**Pull request merged**